src/irc.lua

```diff
@@ -41,12 +56,38 @@ irc.modeset = function (target, recipient, mode)
     end
 end
 
-irc.kick = function (target, recipient, message)
-    if target:byte() == 35 then
-        irc.connection:send(('KICK %s %s :%s\r\n'):format(target, recipient, message))
-    else
-        irc.privmsg(target, 'Cannot kick in query')
-    end
+irc.names = function (channel)
+    irc.connection:send(('NAMES %s\r\n'):format(channel))
+end
+
+irc.nickset = function (nickname)
+    irc.connection:send(('NICK %s\r\n'):format(nickname))
+end
+
+irc.notice = function (target, msg)
+    irc.connection:send(('NOTICE %s :%s\r\n'):format(target, msg))
+end
+
+irc.part = function (channel)
+    irc.connection:send(('PART %s\r\n'):format(channel))
+end
+
+irc.pong = function (sname)
+    irc.connection:send(('PONG %s\r\n'):format(sname))
+end
+
+irc.privmsg = function (target, msg)
+    irc.connection:send(('PRIVMSG %s :%s\r\n'):format(target, msg))
+end
+
+irc.quit = function (msg)
+    irc.connection:send(('QUIT %s\r\n'):format(msg))
+    --irc.connection:close()
```

Comment:
why comment these two lines out?

Reply:
I think when the bot restarts it also quits, and I wasn't sure if you would want to have the bot do /quit when it restarts, but not exist out completely.

Reply:
Another optoins is adding an irc.die function that closes the connection and exits... then you could do irc.die on various events or errors.

Reply:
When the bot fully restarts, it does result in the connection being closed, but it's probably much nicer to the server and for local resource use to make sure we clean-up well. I think, ``close()``ing of the connection is a good idea.